### PR TITLE
Standardise on ExecutionContext

### PR DESCRIPTION
**Pull request by aeons**
Change uses of `ExecutorService` (left over from scalaz days) to `ExecutionContext` (used in cats-effect) to ease the transition in the future.

### CODE REVIEW

File: blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ final case class BlazeClientConfig(// HTTP properties
 
                                    // pipeline management
                                    bufferSize: Int,
-                                   customExecutor: Option[ExecutorService],
+                                   customExecutionContext: Option[ExecutionContext],
```
Comment (rossabaker) [inline blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala]: `customExecutionContext` was so we knew if it was a default, and that we have to shut it down ourselves.  What if we eliminate the option and just use `ExecutionContext.global` as the default?

Comment (aeons) [inline blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala]: Sounds good to me.

File: tomcat/src/main/scala/org/http4s/server/tomcat/TomcatBuilder.scala [hunk 2]
```diff
@@ -180,7 +177,7 @@ object TomcatBuilder extends TomcatBuilder(
   socketAddress = ServerBuilder.DefaultSocketAddress,
   // TODO fs2 port
   // This is garbage how do we shut this down I just want it to compile argh
```
Comment (ChristopherDavenport) [inline tomcat/src/main/scala/org/http4s/server/tomcat/TomcatBuilder.scala]: Can probably remove this now that we are switching.

File: blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala (hunk 1 above)
Comment (ChristopherDavenport) [inline blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientConfig.scala]: :+1:

File: blaze-client/src/main/scala/org/http4s/client/blaze/bits.scala [hunk 3]
```diff
@@ -21,12 +21,13 @@ private[blaze] object bits {
...
-    case Some(exec) => (exec, Task.now(()))
-    case None =>
-      val exec = threads.newDaemonPool("http4s-blaze-client")
-      (exec, Task.delay(exec.shutdown()))
-  }
+  def getExecutionContext(config: BlazeClientConfig): (ExecutionContext, Task[Unit]) =
+    config.customExecutionContext match {
+      case Some(ec) => (ec, Task.now(()))
+      case None =>
+        val exec = threads.newDaemonPool("http4s-blaze-client")
```
Comment (ChristopherDavenport) [inline blaze-client/src/main/scala/org/http4s/client/blaze/bits.scala]: It seems like this should use `org.http4s.util.threads.DefaultExecutionContext` as it is in core so should be available.

**Comment by aeons** [comment]
So, should the default be `ExecutionContext.global` or `org.http4s.util.threads.DefaultExecutionContext`?

**Comment by aeons** [comment]
Also, does this mean that all the stuff in `blaze-client` that takes both a config and an `ExecutionContext` could be changed to just take an `ExecutionContext`?

File: blaze-client/src/main/scala/org/http4s/client/blaze/bits.scala (hunk 3 above)
Comment (ChristopherDavenport) [inline blaze-client/src/main/scala/org/http4s/client/blaze/bits.scala]: So, if I understand correctly. It sounds like standardizing on `ExecutionContext.global` where it is used and getting rid of the option at the edges. 

Then like everything on this branch, get functional first and then we will see where performance improvements can be made.

**Comment by ChristopherDavenport** [comment]
As best I can see, this looks good. Not seeing an obvious hole anywhere. 

:+1:

File: jetty/src/main/scala/org/http4s/server/jetty/JettyBuilder.scala [hunk 4]
```diff
@@ -66,15 +65,14 @@ sealed class JettyBuilder private(
   override def bindSocketAddress(socketAddress: InetSocketAddress): JettyBuilder =
     copy(socketAddress = socketAddress)
 
-  override def withServiceExecutor(serviceExecutor: ExecutorService): JettyBuilder =
-    copy(serviceExecutor = serviceExecutor)
+  override def withExecutionContext(executionContext: ExecutionContext): Self =
```
Comment (rossabaker) [inline jetty/src/main/scala/org/http4s/server/jetty/JettyBuilder.scala]: Actually, one more thing: we could preserve `withServiceExecutor` as a deprecated method, and construct an EC from it and delegate to `withExecutionContext`.  That will help people transition.

**Pull request merged**